### ISSUE TITLE
v5 - Use custom sub-type of Expected losses all custom properties after RunAsync

### ISSUE THREAD

**Issue by HernanFAR** [louthy/language-ext#1340]
When using a subtype of Expected and adding new properties, if you use it in a Effs and it goes to "Wrong way", the custom properties of the subtype are loosing, also it adds a recursive path in "head" property. This in v4 does not happen.

Repository with this code: https://github.com/HernanFAR/language-ext-bug

Example code for v5

```c#
  public class V5Test
  {
      public sealed record CustomExpected(string Message, int Code, string Another) : Expected(Message, Code);

      [Fact]
      public void Test()
      {
        // Arrange
        var expected = new CustomExpected("Name", 100, "This is loss");
        Eff<int> effect = liftEff<int>(() => expected);

        // Act
        Fin<int> fin = effect.Run();

        // Assert
        fin.Match(_ => Assert.True(false),
                  error =>
                  {
                      Assert.Equal(error.Code, expected.Code);
                      Assert.Equal(error.Message, expected.Message);

                      var fail = (CustomExpected)error;
                      Assert.Equal(fail.Another, expected.Another);
                  });

    }
}
```

Example for v4

```c#
public class V4Test
{
    public sealed record CustomExpected(string Message, int Code, string Another)
        : Expected(Message, Code);

    [Fact]
    public void Test()
    {
        // Arrange
        var      expected = new CustomExpected("Name", 100, "This is loss");
        Eff<int> effect   = Eff<int>.Fail(expected);

        // Act
        Fin<int> fin = effect.Run();

        // Assert
        fin.Match(_ => Assert.True(false),
                  error =>
                  {
                      Assert.Equal(error.Code, expected.Code);
                      Assert.Equal(error.Message, expected.Message);

                      var fail = (CustomExpected)error;
                      Assert.Equal(fail.Another, expected.Another);
                  });

    }
}
```

As far as I know, it might be in the ToErrorException part of the exection, but i'm not completedly sure.

Thanks in advance for this library and seeing this issue :)

**Comment by louthy** [comment]
Are you running the latest beta?

**Comment by HernanFAR** [comment]
Yeah! In fact, I find this bug when using the beta1 in my framework, I'll add a link a repository with this tests.

**Comment by HernanFAR** [comment]
Source code added! (Here and in the first comment)
https://github.com/HernanFAR/language-ext-bug

Thanks for seeing this issue!

**Comment by louthy** [comment]
Thanks, I'll take a look 👍

**Comment by louthy** [comment]
This is fixed in `v5.0.0-beta-02`.

**Comment by HernanFAR** [comment]
Fixed! But found another error in #1341

Thanks for all :)
